The generated function FieldMaskWithMode clashes with message field named field_mask_with_mode

The generated function `FieldMaskWithMode` clashes with message field named `field_mask_with_mode`. Eg:

```protobuf
message UserInfoRequest {
  string user_id = 1;
  google.protobuf.FieldMask field_mask = 2 [
    (fieldmask.option.Option).in = {gen: true},
    (fieldmask.option.Option).out = {gen: true, message:"UserInfoResponse"}
  ];
  string field_mask_with_mode = 3;
}
```

```
pb/user.pb.fm.go:11:6: type UserInfoRequest has both field and method named FieldMaskWithMode
```

The solution is to rename generate function as `FieldMask_WithMode` following the pattern with other generated functions.

--- a/examples/pb/user.pb.fm.go
+++ b/examples/pb/user.pb.fm.go
@@ -8,7 +8,7 @@ import (
 	fieldmaskpb "google.golang.org/protobuf/types/known/fieldmaskpb"
 )
 
-func (x *UserInfoRequest) FieldMaskWithMode(mode pbfieldmask.MaskMode) *UserInfoRequest_FieldMask {
+func (x *UserInfoRequest) fieldMaskWithMode(mode pbfieldmask.MaskMode) *UserInfoRequest_FieldMask {
 	fm := &UserInfoRequest_FieldMask{
 		maskMode: mode,
 		mask:     pbfieldmask.New(x.FieldMask),
@@ -17,18 +17,18 @@ func (x *UserInfoRequest) FieldMaskWithMode(mode pbfieldmask.MaskMode) *UserInfo
 	return fm
 }
 
-// FieldMask_Prune generates *UserInfoRequest_FieldMask with filter mode, so that
+// FieldMask_Filter generates *UserInfoRequest_FieldMask with filter mode, so that
 // only the fields in the UserInfoRequest.FieldMask will be
 // appended into the UserInfoRequest.
 func (x *UserInfoRequest) FieldMask_Filter() *UserInfoRequest_FieldMask {
-	return x.FieldMaskWithMode(pbfieldmask.MaskMode_Filter)
+	return x.fieldMaskWithMode(pbfieldmask.MaskMode_Filter)
 }
 
 // FieldMask_Prune generates *UserInfoRequest_FieldMask with prune mode, so that
 // only the fields NOT in the UserInfoRequest.FieldMask will be
 // appended into the UserInfoRequest.
 func (x *UserInfoRequest) FieldMask_Prune() *UserInfoRequest_FieldMask {
-	return x.FieldMaskWithMode(pbfieldmask.MaskMode_Prune)
+	return x.fieldMaskWithMode(pbfieldmask.MaskMode_Prune)
 }
 
 // UserInfoRequest_FieldMask provide provide helper functions to deal with FieldMask.
@@ -55,7 +55,7 @@ func (x *UserInfoRequest) MaskIn_UserId() *UserInfoRequest {
 // exists in the UserInfoRequest.FieldMask or not.
 func (x *UserInfoRequest_FieldMask) MaskedIn_UserId() bool {
 	if x.mask == nil {
-		return false
+		return true
 	}
 
 	return x.mask.Masked("user_id")
@@ -79,7 +79,7 @@ func (x *UserInfoRequest) MaskOut_UserId() *UserInfoRequest {
 // exists in the UserInfoRequest.FieldMask or not.
 func (x *UserInfoRequest_FieldMask) MaskedOut_UserId() bool {
 	if x.mask == nil {
-		return false
+		return true
 	}
 
 	return x.mask.Masked("user_id")
@@ -100,7 +100,7 @@ func (x *UserInfoRequest) MaskOut_Name() *UserInfoRequest {
 // exists in the UserInfoRequest.FieldMask or not.
 func (x *UserInfoRequest_FieldMask) MaskedOut_Name() bool {
 	if x.mask == nil {
-		return false
+		return true
 	}
 
 	return x.mask.Masked("name")
@@ -121,7 +121,7 @@ func (x *UserInfoRequest) MaskOut_Email() *UserInfoRequest {
 // exists in the UserInfoRequest.FieldMask or not.
 func (x *UserInfoRequest_FieldMask) MaskedOut_Email() bool {
 	if x.mask == nil {
-		return false
+		return true
 	}
 
 	return x.mask.Masked("email")
@@ -142,7 +142,7 @@ func (x *UserInfoRequest) MaskOut_Address() *UserInfoRequest {
 // exists in the UserInfoRequest.FieldMask or not.
 func (x *UserInfoRequest_FieldMask) MaskedOut_Address() bool {
 	if x.mask == nil {
-		return false
+		return true
 	}
 
 	return x.mask.Masked("address")
@@ -163,7 +163,7 @@ func (x *UserInfoRequest) MaskOut_Address_Country() *UserInfoRequest {
 // exists in the UserInfoRequest.FieldMask or not.
 func (x *UserInfoRequest_FieldMask) MaskedOut_Address_Country() bool {
 	if x.mask == nil {
-		return false
+		return true
 	}
 
 	return x.mask.Masked("address.country")
@@ -184,7 +184,7 @@ func (x *UserInfoRequest) MaskOut_Address_Province() *UserInfoRequest {
 // exists in the UserInfoRequest.FieldMask or not.
 func (x *UserInfoRequest_FieldMask) MaskedOut_Address_Province() bool {
 	if x.mask == nil {
-		return false
+		return true
 	}
 
 	return x.mask.Masked("address.province")

--- a/internal/templates/go/fm.tpl
+++ b/internal/templates/go/fm.tpl
@@ -2,7 +2,7 @@
 {{ $outMessageName := .OutMessage.Name }}
 {{ $fmField := .FieldMaskField }}
 
-func (x *{{ $inMessageName }}) FieldMaskWithMode(mode pbfieldmask.MaskMode) *{{ $inMessageName }}_FieldMask {
+func (x *{{ $inMessageName }}) fieldMaskWithMode(mode pbfieldmask.MaskMode) *{{ $inMessageName }}_FieldMask {
     fm := &{{ $inMessageName }}_FieldMask{
         maskMode: mode,
         mask: pbfieldmask.New(x.{{$fmField.Name.UpperCamelCase}}),
@@ -11,18 +11,18 @@ func (x *{{ $inMessageName }}) FieldMaskWithMode(mode pbfieldmask.MaskMode) *{{ 
     return fm
 }
 
-// FieldMask_Prune generates *{{ $inMessageName }}_FieldMask with filter mode, so that
+// FieldMask_Filter generates *{{ $inMessageName }}_FieldMask with filter mode, so that
 // only the fields in the {{ $inMessageName }}.{{ $fmField.Name.UpperCamelCase }} will be
 // appended into the {{ $inMessageName }}.
 func (x *{{ $inMessageName }}) FieldMask_Filter() *{{ $inMessageName }}_FieldMask {
-	return x.FieldMaskWithMode(pbfieldmask.MaskMode_Filter)
+	return x.fieldMaskWithMode(pbfieldmask.MaskMode_Filter)
 }
 
 // FieldMask_Prune generates *{{ $inMessageName }}_FieldMask with prune mode, so that
 // only the fields NOT in the {{ $inMessageName }}.{{ $fmField.Name.UpperCamelCase }} will be
 // appended into the {{ $inMessageName }}.
 func (x *{{ $inMessageName }}) FieldMask_Prune() *{{ $inMessageName }}_FieldMask {
-	return x.FieldMaskWithMode(pbfieldmask.MaskMode_Prune)
+	return x.fieldMaskWithMode(pbfieldmask.MaskMode_Prune)
 }
 
 // {{ $inMessageName }}_FieldMask provide provide helper functions to deal with FieldMask.

